packages/api/endpoints/granules.js

```diff
@@ -114,17 +113,12 @@ const create = async (req, res) => {
       dynamoRecord: granule,
       knexOrTransaction: knex,
     });
-    if (
-      await granulePgModel.exists(knex, {
-        granule_id: pgGranule.granule_id,
-        collection_cumulus_id: await collectionPgModel.getRecordCumulusId(
-          knex,
-          deconstructCollectionId(granule.collectionId)
-        ),
-      })
-    ) {
+    // Check if granule already exists across all collections
+    const granulesWithGranuleId = await getGranulesWithGranuleId(knex, pgGranule.granule_id);
+    if (granulesWithGranuleId.length > 0) {
+      log.error('Could not write granule. It already exists.');
       return res.boom.conflict(
-        `A granule already exists for granule_id: ${granule.granuleId}`
+        `A granule already exists for granuleId: ${pgGranule.granule_id}`
```

Comment:
this check does two things:
- check that the granule exists
- disallow granuleIds with a different collectionId